app/scripts/modules/core/authentication/authentication.module.js

```diff
@@ -6,14 +6,20 @@ module.exports = angular.module('spinnaker.authentication', [
   require('./authentication.service.js'),
   require('../config/settings.js'),
   require('./authentication.initializer.service.js'),
-  require('./authentication.interceptor.service.js'),
-  require('./userMenu/userMenu.module.js'),
+  require('../scheduler/scheduler.factory.js')
 ])
   .config(function ($httpProvider) {
     $httpProvider.interceptors.push('gateRequestInterceptor');
   })
-  .run(function (authenticationInitializer, settings) {
-    if(settings.authEnabled) {
+  .run(function (schedulerFactory, authenticationInitializer, settings) {
+    if (settings.authEnabled) {
+      // schedule deck to re-authenticate every 10 min.
+      let authenticationScheduler = schedulerFactory.createScheduler(600000);
+      let reauthenticateUser = () => {
+        authenticationInitializer.reauthenticateUser();
+      };
+      let authenticationLoader = authenticationScheduler.subscribe(reauthenticateUser);
```

Comment:
You should be able to simplify these four lines to:
```javascript
authenticationScheduler.subscribe(authenticationInitializer.reauthenticateUser)
```
If you want to get really really simplified, collapse these five lines to:
```javascript
schedulerFactory.createScheduler(600000).subscribe(authenticationInitializer.reauthenticateUser);
```